tx/signer.go

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) 2024 The VeChainThor developers
+
+// Distributed under the GNU Lesser General Public License v3.0 software license, see the accompanying
+// file LICENSE or <https://www.gnu.org/licenses/lgpl-3.0.html>
+
+package tx
+
+import (
+	"crypto/ecdsa"
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/crypto/secp256k1"
+	"github.com/pkg/errors"
+	"github.com/vechain/thor/v2/thor"
+)
+
+// SignatureFunc is a type alias for a function that signs a hash using a private key.
+// It returns the signature and an error if any occurs.
+type SignatureFunc func(hash []byte, prv *ecdsa.PrivateKey) (sig []byte, err error)
```

Comment:
Do we need this? It seems the signer func is exported, but none of the methods accepting `SignerFunc` are exported.

Could we just use crypto.Sign directly within the functions? Right now there is no use case to pass the signing function

Reply:
Yeah good point, fixed.